app/coffee/UpdatesManager.coffee

```diff
@@ -16,6 +16,15 @@ module.exports = UpdatesManager =
 		if length == 0
 			return callback()
 
+		# check that ops are in the correct order
+		for op, i in rawUpdates when i > 0
+			thisVersion = op?.v
+			prevVersion = rawUpdates[i-1]?.v
+			if not (prevVersion < thisVersion)
+				logger.error project_id: project_id, doc_id: doc_id, rawUpdates:rawUpdates, temporary: temporary, thisVersion:thisVersion, prevVersion:prevVersion, "op versions out of order"
+				# TODO try to recover by sorting the ops
+				return callback(new Error("incoming op versions out of order"))
+
```

Comment:
Is this going to error every time we flush? Can we do like we do with other errors, and mark it as broken, but keep going?